### PR TITLE
Update the EB redistribution

### PR DESCRIPTION
**Pull request by marchdf**
This PR updates the EB redistribution to use IAMR redistribution
utils. The big deal with this PR is that we get to use State
Redistribution, as described by Berger and Giuliani (2021). This is a
better way to perform EB redistribution. I performed extensive
verification of the new method and believe that it works as expected
and performs better than existing redistribution. Specifically:

- we can safely set eb_small_vfrac to zero for all cases
- EB cases perform just as well as existing redistribution, some better
- EB-C7 and EB-C8 no longer need a specific eb_small_vfrac (using 0 works)
- EB-C14 (shock interacting with ramp) runs (does not run with existing)
- EB-C13 is a new case that is added

New options are documented in the code.

This is draft until 
- [x] I finish running C3
- [ ] I finish running C4, C5
- [x] I finish running C12
- [x] I finish running C13
- [x] I make sure all is well on the GPU

This is going to introduce diffs in all the EB cases.

### CODE REVIEW

**Comment by marchdf** [comment]
On EB-C10 I see these diffs GPU vs CPU after 1 step:
```
mhenryde@r104u33 ~/c/P/P/E/R/EB-C10> rm -rf diffs; srun -N 1 ~/combustion/Pele/PeleC/Build/Submodules/AMReX/Tools/Plotfile/fcompare -d Temp plt00001 cpu/plt00001                                                                                                                                                                                                                                                                                                            1 new-redist!?

            variable name            absolute error            relative error
                                        (||A - B||)         (||A - B||/||A||)
 ----------------------------------------------------------------------------
 level = 0
 density                            6.505213035e-19           5.532326894e-16
 xmom                               3.774758284e-15           1.839202667e-15
 ymom                               3.409300836e-15           2.922849829e-11
 zmom                               4.447343101e-15            3.81278058e-11
 rho_E                              1.396983862e-09           5.515477258e-16
 rho_e                              1.396983862e-09           5.519200934e-16
 Temp                               3.410605132e-13           1.136815651e-15
 rho_O2                             6.505213035e-19           5.532326894e-16
 rho_N2                                           0                         0
 pressure                           1.164153218e-09           1.149833528e-15
 kineng                             6.366462912e-12           3.554363704e-15
 enstrophy                          1.826094831e-12           1.369541725e-13
 soundspeed                         2.182787284e-11           6.286860934e-16
 MachNumber                         9.714451465e-17            1.93028159e-15
 entropy                                          0                         0
 magvort                            1.354294454e-11           8.638196658e-14
 divu                               5.712763596e-11           7.210725362e-15
 eint_E                             1.907348633e-06            8.86059705e-16
 eint_e                             2.384185791e-06           1.107574631e-15
 logden                             4.440892099e-16           1.497534373e-16
 x_velocity                         3.410605132e-12           1.952005583e-15
 y_velocity                         2.969319235e-12           2.756322508e-11
 z_velocity                         3.962080664e-12           3.677870666e-11
 magvel                             3.410605132e-12           1.952005583e-15
 radvel                             3.637978807e-12           2.083081646e-15
 magmom                             3.774758284e-15           1.839202667e-15
 vfrac                              1.942890293e-16           1.942890293e-16
```
I call that good enough.

**Comment by marchdf** [comment]
All tests now pass, this is pretty much ready to go.

**Review by jrood-nrel — APPROVED** [review]
I just spent some time looking over this and it's all fine with me.

File: Docs/sphinx/geometry/EB.rst [hunk 1]
```diff
@@ -168,7 +119,7 @@ advanced a single time step, which we refer to as re-redistribution. Specificall
     because the correct distributed mass has to come from the left-coarse-real-cell.
 
 The re-redistribution is implemented as a book-keeping step where the mass distributed are stored during MOL divergence calculation and given to the coarse and fine flux registers to reflux at 
-the end of each time step. The re-redistribution is performed everytime the reflux function is called in post_timestep. More details regarding re-redistribuion are 
+the end of each time step. The re-redistribution is performed everytime the reflux function is called in post_timestep. More details regarding re-redistribution are
```
Comment (drummerdoc) [inline Docs/sphinx/geometry/EB.rst]: A critical issue here.... "everytime" -> "every time"

File: Source/Redistribution/iamr_create_itracker_2d.cpp [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+#if 0
+           if (debug_verbose > 0)
+               amrex::Print() << "Cell " << IntVect(i,j) << " with volfrac " << vfrac(i,j,k) <<
+                                 " trying to merge with " << IntVect(i+ioff,j+joff) <<
+                                 " with volfrac " << vfrac(i+ioff,j+joff,k) <<
+                                 " to get new sum_vol " <<  sum_vol << std::endl;
+#endif
+
+           // If the merged cell isn't large enough, we try to merge in the other direction
+           if (sum_vol < 0.5)
```
Comment (drummerdoc) [inline Source/Redistribution/iamr_create_itracker_2d.cpp:143]: Will this ever be a parameter that will require tuning, or is this a magic number?

**Review by drummerdoc — APPROVED** [review]
Looks good to me

File: Docs/sphinx/geometry/EB.rst (hunk 1 above)
Comment (marchdf) [reply Docs/sphinx/geometry/EB.rst]: good catch

File: Source/Redistribution/iamr_create_itracker_2d.cpp (hunk 2 above)
Comment (marchdf) [reply Source/Redistribution/iamr_create_itracker_2d.cpp:143]: I don't think this will require tuning

**Review by hsitaram — APPROVED** [review]
looks good to me. thanks for changing those "a"s to area. about time!

**Pull request merged**